src/plugins/GAPPSInstall.js

```diff
@@ -0,0 +1,875 @@
+        this.installBtn.classList.add('hidden');
+        this.installedDiv.classList.remove('hidden');
+    }
+
+    checkFileBeforeUpload(file) {
+        if (file) {
+            if (file.name.toLowerCase().endsWith('.apk')) {
+                if (file.size > 900 * 1024 * 1024) {
+                    // 900MB in bytes
+                    this.showUploadError(
+                        this.i18n.FILE_TOO_LARGE ||
+                            `Your file "${file.name}" doesn't respect the conditions to be uploaded (900Mo max).
+                            Please try with another file.`,
+                    );
+                    return;
+                }
+                // If file is valid, hide error and show upload progress
+                this.hideUploadError();
+                this.showUploadProgress(file);
+                this.handleFileUpload(file);
+            } else {
+                this.showUploadError(
+                    this.i18n.FILE_TYPE_NOT_APK ||
+                        `Invalid file type. Only APK files are supported. 
+                        Please select a file with the .apk extension.`,
+                );
+            }
+        } else {
+            this.showUploadError(
+                this.i18n.WRONG_FILE_TYPE_ONLY_FILE_ACCEPTED ||
+                    'Invalid drop content. Please drag and drop a file, not text or other data.',
+            );
+        }
+    }
+
+    handleFileUpload(file) {
+        this.plugin.toolbarBtn.setIndicator('notification');
+        this.plugin.instance.store.dispatch({type: 'DRAG_AND_DROP_UPLOAD_FILE_ENABLED', payload: false});
+        this.plugin.instance.root.classList.add('gm-uploadind-in-progess');
+
+        const msg = {type: 'upload', file};
+        this.fileUploadWorker.postMessage(msg);
+    }
+
+    showUploadProgress(file) {
+        this.hideUploadError();
+        this.uploadProgressSection.classList.remove('hidden');
+        this.uploadFileName.innerHTML = file.name;
+    }
+
+    showUploadError(message) {
+        this.hideUploadProgress();
+        this.uploadErrorSection.classList.remove('hidden');
+        this.uploadErrorText.innerHTML = message;
+    }
+
+    hideUploadProgress() {
+        this.uploadProgressSection.classList.add('hidden');
+    }
+    hideUploadError() {
+        this.uploadErrorSection.classList.add('hidden');
+    }
+
+    disableDragOver() {
+        this.browseButton.disabled = true;
+        this.browseButton.classList.add('disabled');
+        this.dragDropArea.classList.add('disabled');
+        this.removeListerDragAndDropOver?.();
+        this.removeListerDragAndDropLeave?.();
+        this.removeListerDragAndDropDrop?.();
+    }
+
+    enableDragOver() {
+        this.removeListerDragAndDropOver?.();
+        this.removeListerDragAndDropLeave?.();
+        this.removeListerDragAndDropDrop?.();
+
+        this.removeListerDragAndDropOver = this.plugin.instance.addListener(this.dragDropArea, 'dragover', (event) => {
```

Comment:
lister => listener? (multiple places)

Reply:
This looks like quite a bit of code duplication with `FileUpload.js`.

Would it be too complicated to have some common code centralized somewhere?